packages/platform-sdk/test/integration-tests/sdk-v3/error-cases.test.ts

```diff
@@ -0,0 +1,17 @@
+import { apiRootV3 } from '../test-utils'
+
+describe('testing error cases', () => {
+  it('should throw error when a product type is not found', async () => {
+    try {
+      await apiRootV3
+        .productTypes()
+        .withId({ ID: 'non-existing-id' })
+        .get()
+        .execute()
+
+      throw new Error('Should have thrown an error')
```

Comment:
Indeed it does work for me now, however I think this line shouldn't be here, the sdk call chain should be the one throwing the error when the `.execute()` is called, then we can process the error and make our checks in the `.catch()` block.

So aparently, try-catch will not work here because of the async nature of the builder chain, you might want to remove the try-catch and handle the error in the `.cache()`.


```ts
  it('should throw error when a product type is not found', async () => {
    await apiRootV3
      .productTypes()
      .withId({ ID: 'non-existing-id' })
      .get()
      .execute()
      .catch(e => {
        console.log(e, '---------------->')
        expect(e.statusCode).toEqual(404)
      })
  })
```

I tried the above and it worked just fine. So we can restructure the test a bit to include the catch block instead of throwing a generic error in the `try-catch` setup.

Reply:
Thanks for the suggestion. I checked it and there is one problem in your test: if the error is not thrown (the code before my changes), then the test would also pass. This means that it does not test my fix.

Here I explained it a bit better:

```js
it('should throw error when a product type is not found', async () => {
    await apiRootV3
      .productTypes()
      .withId({ ID: 'non-existing-id' })
      .get()
      .execute()
      .then(() => {
          console.log('it should throw an error when ID is non-existing and not reach this console.log')
          fail()  // <--- a method to fail the test
      })
      .catch(e => {
        expect(e.statusCode).toEqual(404)
      })
  })
```

There is usually `fail()` method to fail the test if it reaches a place it shouldn't reach, but it seems to be missing in jest. If you didn‘t encounter it before, it‘s ok and I will leave the test as it is.

Reply:
Since this is an integrated test, I want to believe that the test must fail, since the ID: 'non-existing-id' is actually a non existent ID.